src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/LoggerAdapter.cs

```diff
@@ -90,16 +88,16 @@ public void LogError(string message, params object[] @params)
 
         if (_telemetryReporter is not null)
         {
-            using var _ = DictionaryPool<string, object?>.GetPooledObject(out var props);
+            var properties = new Property[@params.Length + 1];
```

Comment:
It didn't seem worth it since we know the precise length of the array we need.

Reply:
Makes sense to not use it when it doesn't really provide any advantages. Out of curiosity, why doesn't the next file (CompilationTagHelperResolver) just new it up?

Reply:
Great question! :smile:

Reply:
The reason that we don't just new an array up in `CompilationTagHelperResolver` is because we don't want to allocate the array unless we're going to send telemetry, and `telemetryReporter` can be null. Like Roslyn's `TemporaryArray<T>`, `PooledArrayBuilder<T>` doesn't really incur overhead unless items are added to it. However, it makes sense to be clear here and just use an array that we allocate the first time we add a measurement to it.